pkg/polling.go

```diff
@@ -129,6 +133,11 @@ func (bso *PollingObserver) Observe() (types.BlockKey, []types.UpkeepIdentifier,
 			continue
 		}
 
+		if !bso.filterer.Filter()(key) {
```

Comment:
One idea might be to set this up like middleware. For now the only reason we would filter out an upkeep is because it is in a `pending` state. In the future we could want to filter out upkeeps for other reasons such as the api cooldown logic that Chris added to offChain lookup. I'm sure there are more that will come up so it's an idea.